src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+
+enum FilterType {
+  Active = 'Active',
+  Completed = 'Completed',
+  All = 'All',
+}
+
+const filterTodos = (todos: Todo[], filterType: FilterType) => {
+  switch (filterType) {
+    case FilterType.Active:
+      return todos.filter(todo => !todo.completed);
+    case FilterType.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+        setTimeout(() => {
```

Comment:
You can also create `enum` for these messages

Reply:
> You can also create `enum` for these messages

Good point